### PR TITLE
Updated Schnorr Signature specification 

### PR DESCRIPTION
**Pull request by Naviabheeman**
Added rfc6979 based nonce generation algorithm
Corrected formula used to compute e.

### CODE REVIEW

File: doc/tapyrus/schnorr_signature.md [hunk 1]
```diff
@@ -10,14 +10,30 @@ Signing algorithm:
...
-1. Let d' = int(sk).  
-1. Fail if d' = 0 or d' >= n.  
-1. Let P = d'G.  
-1. Let k be the nonce generated using rfc6979.  
-1. Let R = k'G.   
-1. Let k = k' if jacobi(y(R)) = 1, otherwise let k = n - k'.   
-1. Let e = int(sha256(bytes(R) || bytes(P) || m)) mod n.   
-1. The signature is bytes(x(R)) || bytes((k + ed) mod n).   
+1. Let d' = int(sk).
+1. Fail if d' = 0 or d' >= n.
```
Comment (azuchi) [inline doc/tapyrus/schnorr_signature.md]: Here, `n` comes out for the first time, so it is necessary to explain what `n` is.

File: doc/tapyrus/schnorr_signature.md [hunk 2]
```diff
@@ -10,14 +10,30 @@ Signing algorithm:
...
+Let count = 0
+Let nonce = 0
+while nonce = 0   
+    Let nonce = sha256( sk || m || bytes(algorithm))   
+    do count times   
+        nonce = sha256(nonce)   
+    if nonce = 0 or nonce >= n   
+        nonce = 0   
+        count = count + 1   
+return nonce
```
Comment (azuchi) [inline doc/tapyrus/schnorr_signature.md]: The following logic is correct for generation of rfc6979 nonce.
https://tools.ietf.org/html/rfc6979#section-3.2
It is sufficient here to specify that the following additional data is set to "SCHNORR + SHA256".
https://tools.ietf.org/html/rfc6979#section-3.6

Comment (Naviabheeman) [reply doc/tapyrus/schnorr_signature.md]: Its updated now. Is this what you meant?

Comment (azuchi) [reply doc/tapyrus/schnorr_signature.md]: it's good

**Pull request merged**